File: U1/EXAMEN/Comercializadora_Soap_DotNet_GR01/Comercializadora_Soap_DotNet_GR01/Models/DetalleFactura.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Comercializadora_Soap_DotNet_GR01.Models
{
    [Table("DetalleFactura")]
    public class DetalleFactura
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int DetalleId { get; set; }

        [Required]
        public int FacturaId { get; set; }

        [Required]
        public int ProductoId { get; set; }

        [Required]
        public int Cantidad { get; set; }

        [Required]
        public decimal PrecioUnitario { get; set; }

        [Required]
        public decimal Subtotal { get; set; }

        // Navigation properties
        [ForeignKey("FacturaId")]
        public virtual Factura Factura { get; set; }

        [ForeignKey("ProductoId")]
        public virtual Producto Producto { get; set; }
    }
}

File: U2/PAO202551_ELECTRO_BDDNUBE_RESTFULL_JAVA_GR01/04. SERVIDOR/Comercializadora_RestFul_Java_GR01/src/main/java/ec/edu/monster/Models/DetalleFactura.java
package ec.edu.monster.Models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import java.math.BigDecimal;

/**
 * Entidad DetalleFactura
 */
@Entity
@Table(name = "DetalleFactura")
@Data
@NoArgsConstructor
public class DetalleFactura {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "DetalleId")
    private Integer detalleId;

    @Column(name = "Cantidad", nullable = false)
    private Integer cantidad;

    @Column(name = "PrecioUnitario", nullable = false, precision = 12, scale = 2)
    private BigDecimal precioUnitario;

    @Column(name = "Subtotal", nullable = false, precision = 12, scale = 2)
    private BigDecimal subtotal;

    // Relación con Factura
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "FacturaId", nullable = false)
    @ToString.Exclude
    private Factura factura;

    // Relación con Producto
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "ProductoId", nullable = false)
    private Producto producto;
}
